docs/instructor-training-workflow.txt

```diff
@@ -0,0 +1,211 @@
+## Notation
+
+- [use cases] are in square brackets
+- {web pages} are in curly braces
+- <database objects> are in angle brackets
+
+## Individual application for training.
+
+* [user registers self][a]: user match by email address or GitHub ID.
+  - User fills in {single user application form} on AMY with name, email address, GitHub ID, etc.
+  - AMY matches to pre-existing <Person> using email or GitHub ID.
+  - <Training Application> created, connected to pre-existing <Person>, marked "Pending", and added to <Training Application List>.
+
+* [user registers self][b]: no user match.
+  - User fills in form on AMY with name, email address, GitHub ID, etc.
+  - AMY does *not* match email or GitHub ID.
+  - <Training Application> created, marked "Pending", and added to <Training Application List> *without* <Person> object.
```

Comment:
Do I understand correctly that you want `<TrainingApplication>` to have nullable ForeignKey to `<Person>`?